software/control/gui_hcs.py

```diff
@@ -742,6 +706,160 @@ def makeConnections(self):
+            if USE_NAPARI_FOR_MOSAIC_DISPLAY:
+                self.napari_connections['napariMosaicDisplayWidget'] = [
+                    (self.multipointController.napari_mosaic_update, self.napariMosaicDisplayWidget.updateMosaic),
+                    (self.multiPointWidget.signal_acquisition_channels, self.napariMosaicDisplayWidget.initChannels),
+                    (self.multiPointWidget.signal_acquisition_shape, self.napariMosaicDisplayWidget.initLayersShape),
+                    (self.napariMosaicDisplayWidget.signal_coordinates_clicked, 
+                     self.navigationController.move_from_click_mosaic),
+                    (self.napariMosaicDisplayWidget.signal_update_viewer, self.navigationViewer.update_slide)
+                ]
+
+                if ENABLE_FLEXIBLE_MULTIPOINT:
+                    self.napari_connections['napariMosaicDisplayWidget'].extend([
+                        (self.multiPointWidget2.signal_acquisition_channels, self.napariMosaicDisplayWidget.initChannels),
+                        (self.multiPointWidget2.signal_acquisition_shape, self.napariMosaicDisplayWidget.initLayersShape)
+                    ])
+
+                if ENABLE_SCAN_GRID:
+                    self.napari_connections['napariMosaicDisplayWidget'].extend([
+                        (self.multiPointWidgetGrid.signal_acquisition_channels, self.napariMosaicDisplayWidget.initChannels),
+                        (self.multiPointWidgetGrid.signal_acquisition_shape, self.napariMosaicDisplayWidget.initLayersShape),
+                        (self.multiPointWidgetGrid.signal_draw_shape, self.napariMosaicDisplayWidget.enable_shape_drawing),
+                        (self.napariMosaicDisplayWidget.signal_shape_drawn, self.multiPointWidgetGrid.update_manual_shape)
+                    ])
+
+            # Make initial connections
+            self.updateNapariConnections()
+
+    def updateNapariConnections(self):
+        # Update Napari connections based on performance mode. Live widget connections are preserved
+        for widget_name, connections in self.napari_connections.items():
+            if widget_name != 'napariLiveWidget':  # Always keep the live widget connected
+                widget = getattr(self, widget_name, None)
+                if widget:
+                    for signal, slot in connections:
+                        if self.performance_mode:
+                            try:
+                                signal.disconnect(slot)
+                            except TypeError:
+                                # Connection might not exist, which is fine
+                                pass
+                        else:
+                            try:
+                                signal.connect(slot)
+                            except TypeError:
+                                # Connection might already exist, which is fine
+                                pass
+
+    def toggleNapariTabs(self):
```

Comment:
toggle normally implies switching to the opposite state.  Something like `setNapariTabsForPerformanceMode(self)` or soemthing might be better.